Sources/FoundationEssentials/FileManager/FileManager+Directories.swift

```diff
@@ -489,22 +489,10 @@ extension _FileManagerImpl {
     
     var currentDirectoryPath: String? {
 #if os(Windows)
-        var dwLength: DWORD = GetCurrentDirectoryW(0, nil)
-        guard dwLength > 0 else { return nil }
-
-        for _ in 0 ... 8 {
-            if let szCurrentDirectory = withUnsafeTemporaryAllocation(of: WCHAR.self, capacity: Int(dwLength), {
-                let dwResult: DWORD = GetCurrentDirectoryW(dwLength, $0.baseAddress)
-                if dwResult == dwLength - 1 {
-                    return String(decodingCString: $0.baseAddress!, as: UTF16.self)
-                }
-                dwLength = dwResult
-                return nil
-            }) {
-                return szCurrentDirectory
-            }
+        let initialSize = GetCurrentDirectoryW(0, nil)
+        return try? Win32FillNullTerminatedStringBuffer(initialSize: initialSize != 0 ? initialSize : DWORD(MAX_PATH), maxSize: DWORD(Int16.max)) {
```

Comment:
I believe that 5000 should be a reasonable upper bound. NT Paths are limited to 4K I believe (page size), and you need the extra 4 characters for the NT path sigil (`\\?\`).

Reply:
NT paths aren't limited to 4k. Long paths are [documented as being a maximum of 32,767 UTF-16 code units](https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation) and it's completely possible to create a path of such a length. Here's my current directory in PowerShell:

```ps1
PS> (Get-Location).Path.Length
32762
```

32762 plus 4 for `\\?\` plus 1 null terminator = 32767.

I'd rather support up to the absolute maximum the OS allows.